packages/vite/src/node/plugins/oxc.ts

```diff
@@ -159,9 +161,26 @@ export async function transformWithOxc(
 
 export function oxcPlugin(config: ResolvedConfig): Plugin {
   const options = config.oxc as OxcOptions
-  const { jsxInject, include, exclude, ...oxcTransformOptions } = options
+  const {
+    jsxInject,
+    include,
+    exclude,
+    jsxInclude,
+    jsxExclude,
+    ...oxcTransformOptions
+  } = options
 
-  const filter = createFilter(include || /\.(m?ts|[jt]sx)$/, exclude || /\.js$/)
+  const defaultInclude = Array.isArray(include)
+    ? include
+    : [include || /\.(m?ts|[jt]sx)$/]
+  const filter = createFilter(
+    defaultInclude.concat(jsxInclude || []),
+    exclude || /\.js$/,
+  )
+  const jsxFilter = createFilter(
+    jsxInclude || /\.jsx$/,
+    jsxExclude || /\.m?tsx?$/,
```

Comment:
```suggestion
    jsxExclude || /\.(m?[jt]s|tsx)$/,
```
Let's include `.js` here as well, so that when people set `exclude: []`, `.js` won't be treated as JSX by default.